### DISCUSSION TITLE
Can't override router global host option on a per-route level

### DISCUSSION THREAD

**Discussion by HongKilDong** [symfony/symfony#45687] [Q&A]
I want all my routes respond on domain.com, but some specific routes respond only on subdomain.domain.com. 
I tried to set  domain.com as global host value in config/routes/annotations.yaml, like:
```
controllers:
    resource: ../../src/Controller/
    type: annotation
    host: 'domain.com'
```
and for specific controller actions I set custom subdomain as host:
```
class MyController extends AbstractController
{
    /**
     * @Route("/custom", name="custom-subdomain", host="subdomain.domain.com")
     * @throws HttpException
     */
    public function showCustom()
    {
        ....
```
but on `bin/console debug:router custom-subdomain` I get host value `domain.com`, not `subdomain.domain.com`, but if I remove host value from `config/routes/annotations.yaml` on `bin/console debug:router custom-subdomain` I get host value `subdomain.domain.com`. So It's clear that global host value in `config/routes/annotations.yaml` overrides host values set on a per route basis, which is pretty counter-intuitive.

Symfony 5.3.10

**Comment by MrYamous** [comment]
Hi, 

This topic reminds me of other issues I had seen some time ago, [this one](https://github.com/symfony/symfony/issues/43984) and [this other](https://github.com/symfony/symfony/issues/43974).
Does this fit with what you are trying to do? 

If so, it seems that Symfony doesn't allow to do that, at least for the moment.

**Reply by HongKilDong** [reply]
Yes, this clarify things
